app/services/analyze.py

```
@@ -325,15 +228,117 @@ async def _generate_documentation(self,
+        if not doc_texts:
+            logging.warning("No valid documents found for reranking")
+            return results
+        
+        limit = min(len(doc_texts), max(top_k, 1))
+        
+        try:
+            filtered_qs = self._filter_questions(questions)
+            scores_in_doc_idx = await self._cohere_multi_rerank(filtered_qs, doc_texts, limit)
+            scores_by_orig_idx = {indexed_docs[i][0]: s for i, s in scores_in_doc_idx.items()}
+            return self._apply_scores_and_sort(results, scores_by_orig_idx, top_k)
         except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)